src/VisualStudio/Core/Def/NavigateTo/RoslynSearchResultPreviewPanel.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.VisualStudio.Core.Imaging;
+using Microsoft.VisualStudio.Search.Data;
+using Microsoft.VisualStudio.Search.UI.PreviewPanel.Models;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.NavigateTo
+{
+    internal sealed partial class RoslynSearchItemsSourceProvider
+    {
+        /// <summary>
+        /// Roslyn preview for our nav-to result.  We just provide a code-editor with the caret positioned in the
+        /// correct location.
```

Comment:
why do we need to do anything here?  Presumably we could give the location to them and they could preview it?

Reply:
They don't supply this capability automatically.  This is the minimal work you have to do to get preview working.  There's an argument to be made to make this even simpler.  e.g. a factory that just takes "filepath + projectguid + span" and does the above.  @veler for awareness.